src/main/java/com/gmail/kutilandrej/cinema/dao/impl/OrderDaoImpl.java

```diff
@@ -0,0 +1,52 @@
+package com.gmail.kutilandrej.cinema.dao.impl;
+
+import com.gmail.kutilandrej.cinema.dao.OrderDao;
+import com.gmail.kutilandrej.cinema.exception.DataProcessingException;
+import com.gmail.kutilandrej.cinema.lib.Dao;
+import com.gmail.kutilandrej.cinema.model.Order;
+import com.gmail.kutilandrej.cinema.model.User;
+import com.gmail.kutilandrej.cinema.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class OrderDaoImpl implements OrderDao {
+
+    @Override
+    public Order add(Order order) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(order);
+            transaction.commit();
+            return order;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert Order entity", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<Order> getOrderHistory(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            String hql = "FROM Order O JOIN FETCH O.tickets AS T JOIN FETCH T.movieSession "
+                    + "AS MS JOIN FETCH MS.movie JOIN FETCH MS.cinemaHall "
+                    + "JOIN FETCH T.user WHERE O.user = :user";
```

Comment:
Can we do that with fetching only o.tickets?

Reply:
okay